vowpalwabbit/cb_sample.cc

```diff
@@ -0,0 +1,62 @@
+#include "reductions.h"
+#include "cb_sample.h"
+#include "explore.h"
+
+using namespace LEARNER;
+using namespace VW;
+using namespace VW::config;
+
+struct cb_sample_data
+{
+  uint64_t random_seed;
+  uint64_t random_seed_counter;
+};
+
+template <bool is_learn>
+void learn_or_predict(cb_sample_data& data, multi_learner& base, multi_ex& examples)
+{
+  multiline_learn_or_predict<is_learn>(base, examples, examples[0]->ft_offset);
+
+  auto action_scores = examples[0]->pred.a_s;
+
+  uint64_t seed = data.random_seed;
+  if (examples[0]->tag.size() > 0)
+  {
+    if (strncmp(examples[0]->tag.begin(), "seed=", 5) == 0)
+    {
+      substring tag_seed{examples[0]->tag.begin() + 5, examples[0]->tag.begin() + examples[0]->tag.size()};
+      seed = static_cast<uint64_t>(int_of_substring(tag_seed));
```

Comment:
The seed need not be an int.  We should be doing exactly the same manipulations as in the exploration library so the seed is used in the same way.

Reply:
I'll take a look at the explore lib and update this